src/handlers/member.ts

```diff
@@ -93,7 +96,10 @@ export async function removeRole(
       botsHighestRole.id,
       roleID,
     );
-    if (!hasHigherRolePosition) {
+    if (
+      !hasHigherRolePosition &&
+      (await cacheHandlers.get("guilds", guildID))?.ownerID !== botID
```

Comment:
Similarly, we should refactor the awaited function to a new variable, looks ugly this way.